### PR TITLE
[RATE-LIMIT]: RateLimit Helper Functions and Queries

### PR DESCRIPTION
**Pull request by sampocs**
Addresses: #443 
Closes: #441 

## Brief Changelog
* Added RateLimit keeper functions and queries

### CODE REVIEW

File: x/ratelimit/types/keys.go [hunk 1]
```diff
@@ -12,3 +12,11 @@ const (
 	// QuerierRoute defines the module's query routing key
 	QuerierRoute = ModuleName
 )
+
+func KeyPrefix(prefix string) []byte {
+	return []byte(prefix)
+}
+
+const (
+	RateLimitKey = "rate-limit"
```
Comment (antstalepresh) [inline x/ratelimit/types/keys.go:21]: It makes more sense to change `RateLimitKey` to `RateLimitKeyPrefix`

File: x/ratelimit/keeper/rate_limit_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	}
+	return rateLimits
+}
+
+func (s *KeeperTestSuite) TestGetRateLimit() {
+	rateLimits := s.createRateLimits()
+	expectedRateLimit := rateLimits[0]
+
+	actualRateLimit, found := s.App.RatelimitKeeper.GetRateLimit(s.Ctx, expectedRateLimit.Path.Id)
+	s.Require().True(found, "element found")
```
Comment (antstalepresh) [inline x/ratelimit/keeper/rate_limit_test.go:27]: second param for True() should be "element not found" as that param is to show error message.

File: x/ratelimit/keeper/rate_limit.go [hunk 3]
```diff
@@ -1,13 +1,64 @@
...
+}
+
+// Grabs and returns a rate limit object from the store using the PathId
+func (k Keeper) GetRateLimit(ctx sdk.Context, pathId string) (rateLimit types.RateLimit, found bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.RateLimitKey))
+
+	rateLimitKey := types.KeyPrefix(pathId)
+	rateLimitValue := store.Get(rateLimitKey)
+
+	if rateLimitValue == nil || len(rateLimitValue) == 0 {
```
Comment (antstalepresh) [inline x/ratelimit/keeper/rate_limit.go:40]: This checking causes a golang lint error.
`rateLimitValue == nil` is same as `len(rateLimitValue) == 0` so its ok to check only nil value.

**Review by antstalepresh — CHANGES_REQUESTED** [review]
Great work! A few comments here

File: x/ratelimit/keeper/rate_limit_test.go (hunk 2 above)
Comment (sampocs) [reply x/ratelimit/keeper/rate_limit_test.go:27]: discussed in the other PR - once we agree on the best approach, I'll update

**Review by antstalepresh — APPROVED** [review]

**Pull request merged**